app/uk/gov/hmrc/channelpreferences/controllers/PreferenceController.scala

```diff
@@ -84,4 +89,34 @@ class PreferenceController @Inject()(
         }
     }
   }
+
+  def update(key: String, status: String): Action[AnyContent] =
+    Action.async { request =>
+      val correlationId = extractCorrelationId(request)
+      key match {
+        case "itsa"        => handleItsaStatus(status, correlationId)
+        case unexpectedKey => Future.successful(BadRequest(s"The key $unexpectedKey is not supported"))
+      }
+    }
+
+  private def extractCorrelationId(request: Request[AnyContent]): String = {
+    val ACKNOWLEDGEMENT_REFERENCE_MAX_LENGTH = 32
+    val randomId = UUID.randomUUID().toString
+    request.headers
+      .get(CustomHeaders.CorrelationId)
+      .getOrElse(randomId)
+      .replace("-", "")
+      .substring(0, ACKNOWLEDGEMENT_REFERENCE_MAX_LENGTH - 1)
+  }
+
+  private def eisUpdateContact(status: Boolean, correlationId: String): Future[Result] =
```

Comment:
do we need this method? hmm I'm not sure

Reply:
After the refactor - I use it to convert the `Either` into `Result`